src/transformers/models/gpt-j/modeling_gpt-j.py

```diff
@@ -0,0 +1,922 @@
+                pointer = getattr(pointer, "bias")
+            elif scope_names[0] == "wpe" or scope_names[0] == "wte":
+                pointer = getattr(pointer, scope_names[0])
+                pointer = getattr(pointer, "weight")
+            else:
+                try:
+                    pointer = getattr(pointer, scope_names[0])
+                except AttributeError:
+                    logger.info(f"Skipping {'/'.join(name)}")
+                    continue
+            if len(scope_names) >= 2:
+                num = int(scope_names[1])
+                pointer = pointer[num]
+        if name[-1] == "w" and name[-2] in ["out_proj", "k_proj", "q_proj", "v_proj", "c_proj", "c_fc"]:
+            array = array.transpose()
+        if name == ["wte"]:
+            # if vocab is padded, then trim off the padding embeddings
+            array = array[: config.vocab_size]
+        try:
+            assert (
+                pointer.shape == array.shape
+            ), f"Pointer shape {pointer.shape} and array shape {array.shape} mismatched {name}"
+        except AssertionError as e:
+            e.args += (pointer.shape, array.shape)
+            raise
+        print(f"Initialize PyTorch weight {name}")
+        pointer.data = torch.from_numpy(array)
+    # init the final linear layer using word embeddings
+    embs = model.transformer.word_embeddings.weight
+    lin = nn.Linear(embs.size()[1], embs.size()[0], bias=False)
+    lin.weight = embs
+    model.set_output_embeddings(lin)
+    return model
+
+def to_gpu(x, config):
+    colab = False
+    space = False
+    gb = 8000
+    if config.jax:
+        gb = 14500
+    try:
+        if 'google.colab' in str(get_ipython()):
+            colab = True
+    except:
+        pass
+    try:
+        if torch.cuda.get_device_properties(0).total_memory > gb * 1024 * 1024:
+            space = True
+    except:
+        pass
+    if colab or space:
+        return x.cuda()
+    else:
+        return x
+def fixed_pos_embedding(dim=None, seq_len=None):
+    inv_freq = 1. / (10000 ** (torch.arange(0, dim, 2) / dim))
+    sinusoid_inp = torch.einsum('i , j -> i j', torch.arange(seq_len), inv_freq).float()
+    return torch.sin(sinusoid_inp), torch.cos(sinusoid_inp)
+
+Tdef rotate_every_two(x):
+    x1 = x[:, :, :, ::2]
+    x2 = x[:, :, :, 1::2]
+    x = torch.stack((-x2, x1), axis=-1)
+    return rearrange(x, '... d j -> ... (d j)')
+
+def apply_rotary_pos_emb(x, sincos, offset=0):
+    sin, cos = map(lambda t: repeat(t[offset:x.shape[1]+offset,:], "n d -> () n () (d j)", j=2), sincos)
+    return (x * cos) + (rotate_every_two(x) * sin)
+
+def mish(x):
+    return x * torch.tanh(nn.functional.softplus(x))
+
+class GPTJAttentionMixin:
+    """
+    A few attention related utilities for attention modules in GPT Neo, to be used as a mixin.
+    """
+    def _split_heads(self, tensor, num_heads, attn_head_size, rotary):
+        """
+        Splits hidden_size dim into attn_head_size and num_heads
```

Comment:
Since there is no local attention all of this can go in `GPTJSelfAttention` class